agent/unit/manager.go

```diff
@@ -0,0 +1,307 @@
+package unit
+
+import (
+	"sync"
+
+	"golang.org/x/xerrors"
+)
+
+// ErrConsumerNotFound is returned when a consumer ID is not registered.
+var ErrConsumerNotFound = xerrors.New("consumer not found")
```

Comment:
What is a consumer here? This terminology snuck in during development. Should it be `Unit`?